client/dashboard/sites/staging-site-sync-modal/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { Button, Modal, __experimentalText as Text } from '@wordpress/components';
+import { createInterpolateElement, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import InlineSupportLink from '../../components/inline-support-link';
+
+import './style.scss';
+
+interface SyncModalProps {
+	isOpen: boolean;
+	onClose: () => void;
+	syncType: 'pull' | 'push';
+	environment: 'production' | 'staging';
+	siteSlug: string;
+}
+
+const useCopy = ( type: 'pull' | 'push' ) => {
```

Comment:
I think it could be just a simple function `getCopy()`? Not sure the benefit of it being a hook here 🤔 it does not do complex operations.

Reply:
I have originally had a simple function there, but then wanted to match the logic with similar approach we have used for sites syncing modal in the Studio app. That's why I went for the hook there.

But, I agree with your point of view and have simplified the logic to use a simple function. 🙂